src/models/chainspec.rs

```diff
@@ -0,0 +1,470 @@
+use std::collections::HashMap;
+
+use ethereum_types::{H160, H256};
+use serde::Deserialize;
+
+type NodeUrl = String;
+
+#[derive(Debug, Deserialize)]
+struct ChainSpec {
+    name: String,
+    data_dir: String,
+    bootnodes: Vec<NodeUrl>,
+    engine: Engine,
+    hardforks: HardForks,
+    params: Params,
+    genesis: Genesis,
+    precompiles: HashMap<String, HashMap<String, Precompiles>>,
+    balances: HashMap<H160, String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct Engine {
+    name: String,
+    params: EngineParams,
+}
+
+#[derive(Debug, Deserialize)]
+struct EngineParams {
+    period: f32,
```

Comment:
TBH not too happy about using floats. Better change format to have millisecs here. Field should be `Duration`